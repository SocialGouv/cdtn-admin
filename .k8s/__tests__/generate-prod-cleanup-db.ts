//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env";

jest.setTimeout(1000 * 60);
test("kosko generate --prod jobs/cleanup-db", async () => {
  expect(
    await getEnvManifests("prod", "jobs/cleanup-db", {
      ...project("cdtn-admin").prod,
    })
  ).toMatchSnapshot();
});
