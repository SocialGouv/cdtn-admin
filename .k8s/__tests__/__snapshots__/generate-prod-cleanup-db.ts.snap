// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --prod jobs/cleanup-db 1`] = `
"apiVersion: batch/v1beta1
kind: CronJob
metadata:
  name: cleanup-db
  namespace: cdtn-admin
spec:
  schedule: 0 4 * * 6
  jobTemplate:
    spec:
      template:
        spec:
          containers:
            - name: db-cleaner
              image: postgres:10
              command:
                - psql
                - '-c'
                - >
                  -- Delete logged_action older than 3 monthes

                  DELETE FROM audit.logged_actions WHERE action_tstamp_tx <
                  NOW() - INTERVAL '3 month';


                  -- Delete alerts older than 3 monthes

                  DELETE FROM alerts WHERE created_at < NOW() - INTERVAL '3
                  month' AND status IN ('done', 'rejected');
              envFrom:
                - secretRef:
                    name: azure-pg-user
          restartPolicy: Never
"
`;
